frontend/src/slices/articles/actions.ts

```diff
@@ -31,12 +32,25 @@ const fetchOwn = createAsyncThunk<
 
 const createArticle = createAsyncThunk<
   ArticleBaseResponseDto,
-  ArticleRequestDto,
+  {
+    articlePayload: ArticleRequestDto;
+    generatedPrompt: PromptRequestDto | null;
+  },
   AsyncThunkConfig
->(`${sliceName}/create`, (articlePayload, { extra }) => {
-  const { articleApi } = extra;
+>(
+  `${sliceName}/create`,
+  async ({ articlePayload, generatedPrompt }, { extra }) => {
+    const { articleApi, promptApi } = extra;
 
-  return articleApi.create(articlePayload);
-});
+    if (generatedPrompt) {
+      const { id: promptId, genreId } = await promptApi.create(generatedPrompt);
+
+      articlePayload.genreId = genreId;
+      articlePayload.promptId = promptId;
```

Comment:
Let's better make separated call, instead of mutating payload
```suggestion
      return await articleApi.create({
        ...articlePayload,
        genreId,
        promptId
      });
```